ksqldb-engine/src/main/java/io/confluent/ksql/serde/SerdeFeaturesFactory.java

```diff
@@ -87,6 +88,57 @@ public static SerdeFeatures buildValueFeatures(
     return SerdeFeatures.from(builder.build());
   }
 
+  /**
+   * Not all {@code KeyFormat}s are valid internal topic formats. Specifically,
+   * we want to ensure that the key format is (1) not NONE and (2) explicitly sets
+   * the wrapping if it contains only a single column. This method ensures that
+   * both of these are eagerly set.
+   *
+   * @return the key format to use for the output of the aggregation
+   */
+  public static KeyFormat sanitizeKeyFormat(
+      final KsqlConfig ksqlConfig,
+      final KeyFormat keyFormat,
+      final boolean isSingleKey
+  ) {
+    if (keyFormat.getFormatInfo().getFormat().equals(NoneFormat.NAME)) {
+      // explicitly resolve the current default wrapping and write it into the
```

Comment:
I'm confused. Why are we populating a key format in the case where the key format is NONE? This seems like it changes behavior of things like `PARTITION BY NULL` where we in fact want the resulting key format to be NONE...?

Reply:
hmm, let me look into this. I think it might be the same issue as above (I first used this just for aggregates and then I made it fit to everything). I'm not sure why this doesn't cause things to fail.

Reply:
I think it was working because we were always setting the right format in the output anyway and there wasn't any repartition step. I've changed the logic to only "fix" the format on group bys